pages/_document.tsx

```diff
@@ -141,6 +143,13 @@ class CustomDocument extends Document {
               dangerouslySetInnerHTML={{
                 __html: `
               var _paq = window._paq || [];
+              ${
+                isSuperAgent(session)
+                  ? `_paq.push(['setCustomDimension', '1', 'Super-agent connecté']);`
+                  : isAgent(session)
+                  ? `_paq.push(['setCustomDimension', '1', 'Agent connecté']);`
+                  : ''
+              }
```

Comment:
Especially as it only get used to get the dimension. Mayb in metadata have a field `matomoDimension` set and directly use it here

```suggestion
		const matomoDimension = this.props['__NEXT_DATA__']?.props?.pageProps?.metadata?.matomoDimension || null;
		
		...

              `_paq.push(['setCustomDimension', '1', ${matomoDimension}]);`
```